scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/Synchronization.java

```diff
@@ -0,0 +1,1135 @@
+
+    /**
+     * Attempts to compute a mapping for the specified key and its current
+     * mapped value (or {@code null} if there is no current mapping).
+     *
+     * The function is passed as a String containing a groovy closure definition which matches the {@link java.util.function.BiFunction} specification.
+     * The first parameter receives the key, and the second parameter receives the binding associated with this key.
+     *
+     * For example, to either create or append a {@code String} msg to a value
+     * mapping:
+     *
+     * <pre> {@code
+     * store.compute(channel, key, "{ k, v -> (v == null) ? \"" + msg + "\" : v.concat(\"" + msg + "\")}")}</pre>
+     * (Method {@link #merge merge()} is often simpler to use for such purposes.)
+     *
+     * <p>If the function returns {@code null}, the mapping is removed (or
+     * remains absent if initially absent).  If the function itself throws an
+     * (unchecked) exception, the exception is rethrown as a ClosureEvaluationException, and the current mapping
+     * is left unchanged.
+     *
+     * The default implementation is equivalent to performing the following
+     * steps for this {@code map}, then returning the current value or
+     * {@code null} if absent:
+     *
+     * <pre> {@code
+     * V oldValue = map.get(key);
+     * V newValue = remappingFunction.apply(key, oldValue);
+     * if (oldValue != null ) {
+     *    if (newValue != null)
+     *       map.put(key, newValue);
+     *    else
+     *       map.remove(key);
+     * } else {
+     *    if (newValue != null)
+     *       map.put(key, newValue);
+     *    else
+     *       return null;
+     * }
+     * }</pre>
+     *
+     * <p>This implementation guarantees that the operation is performed atomically on the channel map.
+     *
+     * @param channel channel identifier
+     * @param key key with which the specified value is to be associated
+     * @param remappingFunction the function to compute a value
+     * @return the new value associated with the specified key, or null if none
+     * @throws NullPointerException if the specified key is null and
+     *         this map does not support null keys, or the
+     *         remappingFunction is null
+     * @throws CompilationException if an error occurred during the compilation of the specified action
+     * @throws ClosureEvaluationException if an error occurred during the evaluation of the specified action
+     * @throws InvalidChannelException if the channel does not exist
+     * @throws IOException if an error occurred when persisting the modified channel on disk
+     * @since 1.8
+     * @see java.util.Map#compute(Object, BiFunction)
+     * @see java.util.function.BiFunction
+     * @see <a href="http://groovy-lang.org/closures.html">Groovy Closures</a>
+     */
+    Serializable compute(String channel, String key, String remappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)